src/components/library_card_table/LibraryCardTable.tsx

```diff
@@ -0,0 +1,431 @@
+  );
+};
+
+// onClick function to show target card via modal
+const showModalOnClick = (
+    card: LibraryCard,
+    setShowModal: React.Dispatch<React.SetStateAction<boolean>>,
+    setCurrentCard: React.Dispatch<React.SetStateAction<LibraryCard>>
+): void => {
+  setCurrentCard(cloneDeep(card));
+  setShowModal(true);
+};
+
+const LibraryCardTable: React.FC<LibraryCardTableProps> = (props) => {
+  const [libraryCards, setLibraryCards] = useState<LibraryCard[]>(props.libraryCards ?? []);
+  const [tableSize, setTableSize] = useState<number>(10);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [pageCount, setPageCount] = useState<number>(1);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [filteredCards, setFilteredCards] = useState<LibraryCard[]>([]);
+  const [visibleCards, setVisibleCards] = useState<LibraryCard[]>([]);
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [showConfirmation, setShowConfirmation] = useState<boolean>(false);
+  const [users, setUsers] = useState<UserData[]>(props.users ?? []);
+  const [currentCard, setCurrentCard] = useState<LibraryCard>({} as LibraryCard);
+  const searchRef = useRef<HTMLInputElement>(null);
+
+  const columnHeaderData: ColumnHeader[] = [
+    columnHeaderFormat.researcher,
+    columnHeaderFormat.email,
+    columnHeaderFormat.createDate,
+    columnHeaderFormat.actions,
+  ];
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        setPageCount(calcTablePageCount(tableSize, filteredCards));
+        if (currentPage > pageCount) {
+          setCurrentPage(pageCount);
+        }
+        const visibleList = calcVisibleWindow(
+            currentPage,
+            tableSize,
+            filteredCards
+        );
+        setVisibleCards(visibleList);
+      } catch (error: unknown) {
+        const axiosError = error as AxiosError;
+        const consentError = axiosError?.response?.data as ConsentError;
+        const serverError = consentError.message ?? 'Error updating Library Card table';
+        Notifications.showError({text: serverError});
+      }
+    };
+    init();
+  }, [filteredCards, tableSize, currentPage, pageCount]);
+
+  // Hook to execute on initialization and card creation/deletion, applies filter on updated collection list
+  useEffect(() => {
+    if (searchRef.current) {
+      const searchTerms = searchRef.current.value;
+      let filteredList = libraryCards;
+      if (!isEmpty(searchTerms)) {
+        filteredList = lcFilterFunction(searchRef, libraryCards);
+      }
+      setFilteredCards(filteredList);
+    }
+  }, [props.libraryCards, libraryCards]);
+
+  // Hook that executes on prop load (initialization hook)
+  useEffect(() => {
+    setLibraryCards(props.libraryCards ?? []);
+    setUsers(props.users ?? []);
+    if (
+        !isNil(props.libraryCards) &&
+        !isNil(props.users)
+    ) {
+      setIsLoading(false);
+    }
+  }, [props.libraryCards, props.users]);
```

Comment:
Do we need this hook? We are already setting these values in `setState` above. If we don't need this hook, then we can probably simplify this further and may not even need `setLibraryCards` / `setUsers`.

Reply:
This is almost an exact [copy of the JSX version](https://github.com/DataBiosphere/duos-ui/pull/2883/files#diff-42fc947cca7b1cb3004ad08ad36c5ac776be3898172cdcdee1e3efeb8bff204dL243) so I didn't look into trying to remove it. I can dig in a little further and try to do that.

Reply:
I tested this out, we do need this - none of the table data loads if we don't call this hook.